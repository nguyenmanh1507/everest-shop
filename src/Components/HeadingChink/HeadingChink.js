// @flow

import React, { Component } from 'react'
import cx from 'classnames'

type Props = {
  /** Heading title */
  title: string,
  /** Extra class */
  exClass?: string
}

class HeadingChink extends Component<Props> {
  render() {
    const { title, exClass } = this.props

    return (
      <h2
        className={cx('text-uppercase text-large', {
          'mb-0': !exClass,
          [exClass || '']: exClass && exClass.length > 0
        })}
      >
        {title}
      </h2>
    )
  }
}

export default HeadingChink
