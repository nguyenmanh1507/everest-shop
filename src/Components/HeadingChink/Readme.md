HeadingChink example

```js
<HeadingChink title="Related products" />
```
