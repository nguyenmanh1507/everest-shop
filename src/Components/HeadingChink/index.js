// @flow

export { default } from './HeadingChink'
