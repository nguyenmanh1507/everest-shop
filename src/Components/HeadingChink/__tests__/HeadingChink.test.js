// @flow

import React from 'react'
import renderer from 'react-test-renderer'

import HeadingChink from '../HeadingChink'

test('HeadingChink render properly', () => {
  const component = renderer.create(<HeadingChink title="related products" />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('HeadingChink with extra class', () => {
  const component = renderer.create(
    <HeadingChink title="related products" exClass="mb-lg" />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
