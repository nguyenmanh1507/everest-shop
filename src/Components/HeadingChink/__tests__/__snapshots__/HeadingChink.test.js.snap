// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingChink render properly 1`] = `
<h2
  className="text-uppercase text-large mb-0"
>
  related products
</h2>
`;

exports[`HeadingChink with extra class 1`] = `
<h2
  className="text-uppercase text-large mb-lg"
>
  related products
</h2>
`;
